src/Exception/GuzzleException.php

```diff
@@ -1,13 +1,29 @@
 <?php
 namespace GuzzleHttp\Exception;
 
-/**
- * @method string getMessage()
- * @method \Throwable|null getPrevious()
- * @method mixed getCode()
- * @method string getFile()
- * @method int getLine()
- * @method array getTrace()
- * @method string getTraceAsString()
- */
-interface GuzzleException {}
+use Throwable;
+
+if (interface_exists(Throwable::class)) {
+    /**
+     * @method string getMessage()
+     * @method Throwable|null getPrevious()
+     * @method mixed getCode()
+     * @method string getFile()
+     * @method int getLine()
+     * @method array getTrace()
+     * @method string getTraceAsString()
```

Comment:
these annotations should not be needed anymore because the method signatures are provided by [throwable](http://php.net/manual/en/class.throwable.php)